velox/functions/prestosql/window/AggregateWindow.cpp

```diff
@@ -113,16 +115,46 @@ class AggregateWindowFunction : public exec::WindowFunction {
     auto rawFrameEnds = frameEnds->as<vector_size_t>();
 
     FrameMetadata frameMetadata =
-        fillArgVectors(rawFrameStarts, rawFrameEnds, numRows);
+        computeFrameMetadata(rawFrameStarts, rawFrameEnds, numRows);
+
     if (frameMetadata.fixedFirstRow && frameMetadata.nonDecreasingFrameEnd) {
+      // If the previous block also had the same firstRow as the current block
+      // and has non-decreasing frames, then aggregation can be resumed from the
+      // results of the previous block. Else, a new aggregation has to be
+      // started.
+      bool usePreviousAggregate = false;
```

Comment:
Consider moving this boolean into FrameMetadata and the logic to compute it to computeFrameMetadata. Also, consider moving the check into a method in FrameMetadata.